rbc/remotejit.py

```diff
@@ -678,30 +817,38 @@ def remote_compile(self, func, ftype: Type, target_info: TargetInfo):
         """
         if self.debug:
             print(f'remote_compile({func}, {ftype})')
-        llvm_module, succesful_fids = irtools.compile_to_LLVM(
-            [(func, {0: ftype})],
-            target_info,
-            pipeline_class=OmnisciCompilerPipeline,
-            debug=self.debug)
+        with target_info:
+            llvm_module, succesful_fids = irtools.compile_to_LLVM(
+                [(func, {0: ftype})],
+                target_info,
+                pipeline_class=OmnisciCompilerPipeline,
+                debug=self.debug)
         ir = str(llvm_module)
         mangled_signatures = ';'.join([s.mangle() for s in [ftype]])
         response = self.client(remotejit=dict(
             compile=(func.__name__, mangled_signatures, ir)))
         assert response['remotejit']['compile'], response
         return llvm_module
 
-    def remote_call(self, func, ftype: Type, arguments: tuple):
+    def remote_call(self, func, ftype: Type, arguments: tuple, try_run=False):
         """Call function remotely on given arguments.
 
         The input function `func` is called remotely by sending the
         arguments data to remote host where the previously compiled
         function (see `remote_compile` method) is applied to the
         arguments, and the result is returned to local process.
+
+        If `try_run` is True then return an object that specifies
+        remote call but does not execute it. The type of return object
+        is custom to particular RemoteJIT specialization.
```

Comment:
`try_run` is partially supported in RemoteJIT.

The `try_run` feature is internally used in composition of remote functions in order to avoid transferring intermediate results between the client and the server. In this sense, both `try_run` and `delay_call` are a little misleading. Perhaps `postpone_execution` or similar?

The feature requires more work for RemoteJIT as currently the intermediate results are transferred..

Reply:
Using postpone_execution.